lib/engine/route.rb

```diff
@@ -4,8 +4,8 @@
 
 module Engine
   class Route
-    attr_accessor :halts, :routes
-    attr_reader :last_node, :phase, :train, :abilities
+    attr_accessor :halts, :routes, :train
```

Comment:
when the train is set do you need to clear the cache?

Reply:
No, I don't think it's required, but it can't hurt either.

Reply:
in that case remove the accessor and do

def train=(new_train)
   ...

Reply:
OK